### DISCUSSION TITLE
Multiple wiring options

### DISCUSSION THREAD

**Discussion by cybercop23** [xLightsSequencer/xLights#4678] [Ideas]
We recently discovered that the vendor model xml file supports up to 4 wiring options for different wiring options (assuming left/right, new wiring/no splice etc). Seems that nobody uses it or even knows about it, since none of the vendors ever have any model with more than one wiring diagram. If they happen to have 2 versions or newer wiring diagram, they would have 2 models.
SO....
Can we (with some changes) repourpose this option to allow for multiple automagic imports? Think mini-tree with native star, or 8 MHs with the 2 dimmer/shutter single line.

Thoughts/ideas?

**Comment by keithsw1111** [comment]
No ... and i think multi model import in a single model already exists as well.

**Reply by cybercop23** [reply]
That's great, but how does it work? We tried multi models in one xmodel and that fails on import. I don't see (maybe I missed it) any tags/code that will allow that?

**Comment by cybercop23** [comment]
@keithsw1111 I found how Ultimesh creates a custom model with embeded shadow models, but can we do a custom or native tree with a star, all as one downloaded xmodel file?

**Comment by keithsw1111** [comment]
Done.

<?xml version="1.0" encoding="UTF-8"?>
<treemodel 
AlternateNodes="false" NoZig="false" StrandDir="Vertical" name="Tree" parm1="16" parm2="50" parm3="1" DisplayAs="Tree 360" StringType="RGB Nodes" Transparency="0" PixelSize="2" ModelBrightness="0" Antialias="1" StartSide="B" Dir="L" StrandNames="" NodeNames="" TreeRotation="3" TreeBottomTopRatio="6.0" TreePerspective="0.2" TreeSpiralRotations="0.0" SourceVersion="2024.13"  >
	<associatedModels>
		<starmodel 
		name="Star" parm1="1" parm2="50" parm3="5" DisplayAs="Star" StringType="RGB Nodes" Transparency="0" PixelSize="2" ModelBrightness="0" Antialias="1" LayerSizes="" starRatio="2.618034" StarStartLocation="Bottom Ctr-CW" StrandNames="" NodeNames="" SourceVersion="2024.13"  >
		</starmodel>
	</associatedModels>
</treemodel>

**Reply by cybercop23** [reply]
Sweet. Thanks so much.
